### PR TITLE
feature: hide vfolder sharing when invite others disabled

### LINKED ISSUE #1629: Let's add option to hide sharing virtual folder
### Main idea

There are certain situations that don't need sharing virtual folders (e.g. shortage of storage...)
For now, we've been providing sharing virtual folders as a default, but we can convert it as an "optional" feature.

### Alternative ideas

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by lizable**
## Description
This PR resolves #1629.

> NOTE: you may need to change `allowed_vfolder_hosts` values to test the visibility of icon-button related to `vfolder sharing`. They will show-up when `invite-others` are allowed at least in one of the policies(`domain`, `project(group)`, `keypair resource policy`).

## Screenshot(s)
Screenshot 2023-03-17 at 8 31 22 PM

### CODE REVIEW

File: src/components/backend-ai-storage-list.ts [hunk 1]
```diff
@@ -1648,15 +1669,18 @@ export default class BackendAiStorageList extends BackendAIPage {
...
         this.authenticated = true;
         this._APIMajorVersion = globalThis.backendaiclient.APIMajorVersion;
         this._maxFileUploadSize = globalThis.backendaiclient._config.maxFileUploadSize;
-        this._checkFilebrowserSupported();
-        this._refreshFolderList(false, 'viewStatechanged');
+        await this._checkFilebrowserSupported();
+        await this._getVolumeInformation();
+        await this._getAllowedVFolderHostsByCurrentUserInfo();
+        await this._triggerFolderListChanged();
+        await this._refreshFolderList(false, 'viewStatechanged');
```
Comment (adrysn) [inline src/components/backend-ai-storage-list.ts]: Is there a reason to `await` every method? Due to these lines, the folder list items are displayed after some white screen. Especially, when I clicked the "AUTOMOUNT FOLDERS", it displayed no items for 2-3 seconds even though I have some dot folders.

**Review by adrysn — COMMENTED** [review]
The initial response time to list the vfolder items is increased in this PR.

File: src/components/backend-ai-storage-list.ts [hunk 2]
```diff
@@ -1648,15 +1669,18 @@ export default class BackendAiStorageList extends BackendAIPage {
...
+      document.addEventListener('backend-ai-connected', async () => {
         this.is_admin = globalThis.backendaiclient.is_admin;
         this.enableStorageProxy = globalThis.backendaiclient.supports('storage-proxy');
         this.enableInferenceWorkload = globalThis.backendaiclient.supports('inference-workload');
         this.authenticated = true;
         this._APIMajorVersion = globalThis.backendaiclient.APIMajorVersion;
         this._maxFileUploadSize = globalThis.backendaiclient._config.maxFileUploadSize;
-        this._checkFilebrowserSupported();
-        this._refreshFolderList(false, 'viewStatechanged');
+        await this._checkFilebrowserSupported();
```
Comment (adrysn) [inline src/components/backend-ai-storage-list.ts]: This query takes a long time if there are several hundreds of images.

Comment (adrysn) [reply src/components/backend-ai-storage-list.ts]: Resolved this issue by fetching the installed images only.

File: src/components/backend-ai-storage-list.ts (hunk 1 above)
Comment (adrysn) [reply src/components/backend-ai-storage-list.ts]: I removed the `await` keywords here after some testing. Please let me know if it has a side-effect.

**Comment by adrysn** [comment]
@lizable I refactored some parts of the code. Please review and test it if available.

**Comment by lizable** [comment]
> jihyunkang (@lizable) I refactored some parts of the code. Please review and test it if available.

I just checked the code and did the E2E test by including/excluding `invite-others` permission, and It works fine.
I'll merge it if there's no other requests related to this PR.

**Pull request merged**